### PR TITLE
Exclude actions not allowed to use outside

### PR DESCRIPTION
**Pull request by vineeth-asml**
1. Action data populates all the action repos that is accessible to the used token
Filter the data based on below conditions
1. If repo visibility is private - Don't add to dataset
2. If internal repo
      - [Get the level of access for workflows outside of the repository](https://docs.github.com/en/rest/actions/permissions#get-the-level-of-access-for-workflows-outside-of-the-repository)
      - If access_level is none - Don't add to the data set

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -151,8 +153,26 @@ async function findAllActions(
     const content = await getActionFile(client, repo)
     if (content && content.name !== '') {
       core.info(
-        `Found action file in repository: ${repo.name} with filename [${content.name}] download url [${content.downloadUrl}]`
+        `Found action file in repository: ${repo.name} with filename [${content.name}] download url [${content.downloadUrl}]. Visibility of repo is ${repo.visibility}`
```
Comment (rajbos) [inline src/main.ts:156]: Let's follow the convention here like in the rest of the repo:
```suggestion
        `Found action file in repository: [${repo.name}] with filename [${content.name}] download url [${content.downloadUrl}]. Visibility of repo is [${repo.visibility}]`
```

File: src/main.ts [hunk 2]
```diff
@@ -151,8 +153,26 @@ async function findAllActions(
...
       )
+
+      // Check and exclude the Work in progress actions if its internal
+      if (repo.visibility == 'internal') {
+        core.debug(`Get Access settings for repository ${repo.owner}/${repo.name}..............`)
```
Comment (rajbos) [inline src/main.ts:161]: Nitpicking
```suggestion
        core.debug(`Get access settings for repository ${repo.owner}/${repo.name}..............`)
```

File: src/main.ts [hunk 3]
```diff
@@ -151,8 +153,26 @@ async function findAllActions(
...
+        const { data: accessSettings } = await client.rest.actions.getWorkflowAccessToRepository({
+          owner: repo.owner,
+          repo: repo.name,
+        })
+        if (accessSettings.access_level == 'none') {
+          core.info(`Access to use action [${repo.owner}/${repo.name}] is disabled`)
+          continue
+        }
+      } else if (repo.visibility == 'private') {
+        core.info(`${repo.owner}/${repo.name} is private repo.`)
```
Comment (rajbos) [inline src/main.ts:172]:
```suggestion
        core.info(`[${repo.owner}/${repo.name}] is a private repo`)
```

**Review by rajbos — CHANGES_REQUESTED** [review]
Looking good! Some small changes

File: src/main.ts (hunk 1 above)
Comment (vineeth-asml) [reply src/main.ts]: Updated with changes

**Review by rajbos — APPROVED** [review]

**Comment by vineeth-asml** [comment]
Thanks @rajbos . You can merge it as well since i don't have write access on the repo

**Comment by rajbos** [comment]
Thanks for adding this!

**Pull request merged**